sdl-kafka/src/main/scala/io/smartdatalake/workflow/dataobject/KafkaTopicDataObject.scala

```diff
@@ -481,22 +527,48 @@ private case class TopicPartitionOffsets(topicPartition: TopicPartition, startOf
     Seq.fill(size)(this.copy(startOffset = endOffset))
   }
 
+  def getStartOffsetForSpark: String = getOffsetForSpark(topicPartition.partition, startOffset, TopicPartitionOffsets.defaultOffsetEarliest)
```

Comment:
Good hint. I would delegate this to the system managing the state - in case of KafkaStateIncrementalMode the current offsets for a consumer group are managed by Kafka. You can use `kafka-consumer-groups` tool to change the current offsets, see also https://stackoverflow.com/questions/29791268/how-to-change-start-offset-for-topic.
In case of DataObjectStateIncrementalMode you would need to manipulate SDLB state files.